pptx_blueprint/__init__.py

```diff
@@ -33,14 +33,47 @@ def replace_text(self, label: str, new_text: str) -> None:
         for shape in shapes:
             shape.text = new_text
 
-    def replace_picture(self, label: str, filename: _Pathlike) -> None:
+    def replace_picture(self, label: str, filename: _Pathlike, *, do_not_scale_up: bool = False) -> None:
```

Comment:
What is the motivation / usecase szenario for `do_not_scale_up`?

I would want the  following behaviors in the future (might be in a future PR).
~~~
scale:
    All image scaling preseves the aspect-ratio of the original image.
    - 'fit-width': Scale the image to the width of the placeholder.
    - 'fit-height': Scale the image to the height of the placeholder.
    - 'fit': Scale the image to fit inside the placeholder.
~~~

Note, that `do_not_scale_up` and `scale` are sort of orthogonal.

Naming: One might want to use `scale_policy` instead of `do_not_scale_up`; with values "shrink-only", "enlarge-only", `None`.
~~~
scale_policy

Reply:
With my implementation the image does always fit into the size of the replaced shape. By calculating the aspect ratio I am able to defect if `height` or `width` is the limiting dimension. This is the same as `scale='fit'`.
I don't like the idea of adding the options 'fit-width' and 'fit-height'. Allowing the image to get larger than the placeholder will result in overlapping with other shapes.
The `scale_up=False` is useful if an image is only a little to small. Scaling it up by a view percent will reduce the image quality. By preventing up scaling we will get the best image quality.

Reply:
> I don't like the idea of adding the options 'fit-width' and 'fit-height'. Allowing the image to get larger than the placeholder will result in overlapping with other shapes.

Not saying that they should be default. But we shouldn't assume too much on the user's intention. I can imagine situations in which you'd really want to force the width/height - no matter the cost.

I agree that preventing upscaling might be desirable. Similarly, somebody might want to prevent downscaling since it reduces the information in the image.